src/pcapi/core/offers/api.py

```diff
@@ -111,75 +114,107 @@ def create_offer(offer_data: PostOfferBodyModel, user: User) -> Offer:
     subcategory = subcategories.ALL_SUBCATEGORIES_DICT.get(offer_data.subcategory_id)
     venue = load_or_raise_error(Venue, offer_data.venue_id)
     check_user_has_access_to_offerer(user, offerer_id=venue.managingOffererId)
+    _check_offer_data_is_valid(offer_data)
     if offer_data.product_id:
-        product = load_or_raise_error(Product, offer_data.product_id)
-        product_subcategory = subcategories.ALL_SUBCATEGORIES_DICT[product.subcategoryId]
-        offer = Offer(
-            product=product,
-            type=product_subcategory.matching_type,  # FIXME: fseguin(2021-07-22): deprecated
-            subcategoryId=product.subcategoryId,
-            name=product.name,
-            description=product.description,
-            url=product.url,
-            mediaUrls=product.mediaUrls,
-            conditions=product.conditions,
-            ageMin=product.ageMin,
-            ageMax=product.ageMax,
-            durationMinutes=product.durationMinutes,
-            isNational=product.isNational,
-            extraData=product.extraData,
-        )
-    elif FeatureToggle.ENABLE_ISBN_REQUIRED_IN_LIVRE_EDITION_OFFER_CREATION.is_active() and can_create_from_isbn(
-        subcategory_id=subcategory.id if subcategory else None, offer_type=subcategory.matching_type
-    ):
-        product = _load_product_by_isbn_and_check_is_gcu_compatible_or_raise_error(offer_data.extra_data["isbn"])
-        product_subcategory = subcategories.ALL_SUBCATEGORIES_DICT[product.subcategoryId]
-        extra_data = product.extraData
-        extra_data.update(offer_data.extra_data)
-        offer = Offer(
-            product=product,
-            type=product_subcategory.matching_type,  # FIXME: fseguin(2021-07-22): deprecated
-            subcategoryId=product.subcategoryId,
-            name=offer_data.name,
-            description=offer_data.description if offer_data.description else product.description,
-            url=offer_data.url if offer_data.url else product.url,
-            mediaUrls=offer_data.url if offer_data.url else product.url,
-            conditions=offer_data.conditions if offer_data.conditions else product.conditions,
-            ageMin=offer_data.age_min if offer_data.age_min else product.ageMin,
-            ageMax=offer_data.age_max if offer_data.age_max else product.ageMax,
-            isNational=offer_data.is_national if offer_data.is_national else product.isNational,
-            extraData=extra_data,
-        )
+        offer = create_product_offer(offer_data)
```

Comment:
```suggestion
        offer = create_offer_from_template(offer_data)
```
Nous n'avons pas d'"offre produit"
Les `products` servent pour garder des valeurs transversent (genre lorsqu'on importe les livres de titelive, et qu'on souhaite créer automatiquement un livre à partir de son isbn

Reply:
C'est pas un `create_offer_from_product` dans l'idée ?